mobile/lib/components/ramowka.dart

```diff
@@ -0,0 +1,398 @@
+                splashColor: context.colors.highlightGreen.withOpacity(0.3),
+                radius: 100,
+                child: Row(
+                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                  children: [
+                    GestureDetector(
+                      // TODO: nagivation to Ramowka page
+                      onTap: () {
+                        // print("Tapped Ramowka");
+                      },
+                      child: Text(
+                        'Ramówka',
+                        style: context.textStyles.textMedium,
+                      ),
+                    ),
+                    GestureDetector(
+                      // TODO: nagivation to Ramowka page
+                      onTap: () {
+                        // print("Clicked button");
+                      },
+                      child: Icon(
+                        Icons.menu,
+                        size: 28,
+                        color: context.colors.highlightGreen,
+                      ),
+                    ),
+                  ],
+                ),
+              ),
+            ),
+          ),
+        ),
+        child: RamowkaList(
+          timeout: timeout,
+          dataSource: dataSource,
+        ),
+      ),
+    );
+  }
+}
+
+/// List of Ramowka entries.
+class RamowkaList extends StatefulWidget {
+  const RamowkaList({
+    super.key,
+    required this.timeout,
+    this.rows = 7,
+    this.rowHeight = 22.0,
+    this.dataSource,
+  });
+
+  /// Timeout for the fetching function.
+  final Duration timeout;
+
+  /// Number of rows
+  final int rows;
+
+  /// Single row's height
+  final double rowHeight;
+
+  /// Optional source of the data to fill
+  /// the list, pulled from the radioaktywne.pl api on default.
+  final Future<void>? dataSource;
+
+  @override
+  State<RamowkaList> createState() => _RamowkaListState();
+}
+
+class _RamowkaListState extends State<RamowkaList> {
+  Future<void>? _ramowkaFuture;
+  var _ramowka = <RamowkaInfo>[];
+  static final Uri _url = Uri.parse(
+    'https://radioaktywne.pl/wp-json/wp/v2/event?_embed=true&page=1&per_page=100',
+  );
+
+  @override
+  void initState() {
+    _ramowkaFuture = widget.dataSource ?? Future<void>(_updateRamowka);
+    super.initState();
+  }
+
+  Future _updateRamowka() async {
+    /// Debug: you can plug external data source in to test the widget
+    if (widget.dataSource != null) {
+      return;
+    }
+
+    final response = await http.get(
+      _RamowkaListState._url,
+      headers: {'Content-Type': 'application/json'},
+    ).timeout(widget.timeout);
+    final jsonData = jsonDecode(response.body) as List<dynamic>;
+    final currentTime =
+        DateFormat(DateFormat.HOUR24_MINUTE).format(DateTime.now());
+
+    final data = jsonData.map(
+      (dynamic data) => RamowkaInfo.fromJson(data as Map<String, dynamic>),
+    );
+
+    setState(
+      () {
+        _ramowka = data
```

Comment:
extract to a method with name indicating what is going on smh like "checkGivenHour()" or "selectRamowkaItems", absolutely no idea what is going on at the first glance 😅

Reply:
I agree, there really was too much clutter